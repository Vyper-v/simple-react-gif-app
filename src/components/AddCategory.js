import React,{useState} from 'react';
import PropTypes from 'prop-types';


function AddCategory({ setCategories }) {
  const [value, setValue] = useState('');
  function handleInputChange(e) {
    setValue(e.target.value);
  }

  function handleSubmit(e) {
    e.preventDefault();
    if (value !== ''){
      setCategories(categs => [value,...categs]);
      setValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit} className='flex flex-col'>
      <input className="p-1 outline-1" type="text" placeholder='Type something...' value={value} onChange={handleInputChange} />
    </form>
  );
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};


export default AddCategory;
