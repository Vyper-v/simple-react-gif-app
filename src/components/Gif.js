import React from 'react';
import PropTypes from 'prop-types';

const Gif = ({title,url}) => {
  return <div className='flex flex-col justify-center items-baseline rounded overflow-hidden shadow shadow-lime-500 bg-lime-200 text-lime-700 transition-all'>
    <img className='object-cover w-full h-[20rem]' src={url} alt={title} />
    <span className='text-lg p-4 flex-auto'>{title}</span>
  </div>;
};

Gif.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Gif;
