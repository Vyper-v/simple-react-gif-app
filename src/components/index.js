export {default as Gif} from './Gif'
export {default as GifLayout} from './GifLayout'
export {default as AddCategory} from './AddCategory'