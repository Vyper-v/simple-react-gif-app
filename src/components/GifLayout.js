import PropTypes from 'prop-types';

import Gif from './Gif';
import { useFetchGifs } from '../hooks/useFechGifs';
import './GifLayout.css'

const GifLayout = ({category}) => {
  const {data:images,loading} = useFetchGifs(category);

  return (
  <div className='grid autoLayout gap-2'>
    { loading && <p className='text-center'>Loading...</p> }
    { images.map( (img) =>  <Gif key={img.id} {...img}/> ) }
  </div>
  );
};

GifLayout.propTypes = {
  category: PropTypes.string.isRequired
};

export default GifLayout;
