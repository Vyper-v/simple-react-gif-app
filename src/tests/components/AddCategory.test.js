import { AddCategory } from "../../components";
import {shallow} from 'enzyme';


const simulateInputChange = (wrapper,inputSelector,newValue) =>{
  const input = wrapper.find(inputSelector);
  input.simulate('change',{target:{value:newValue}})
  return wrapper.find(inputSelector);
}

describe('AddCategory Component tests', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories}/>)

  beforeEach(()=>{
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories}/>);
  })

  test('should be rendered correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('input field should be change', () => {
    const value = "Hello";
    const inputValue = simulateInputChange(wrapper,'input',value);
    expect(inputValue.prop('value')).toBe(value);
  });

  test('setCategories should not be called without input', () => {
    // simulateInputChange(wrapper,'input',""); reset input
    wrapper.simulate('submit',{ preventDefault(){} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('setCategories should be called and cleared the input', () => {
    const testValue = "overwatch";
    let input = simulateInputChange(wrapper,'input',testValue);
    wrapper.simulate('submit',{preventDefault(){}})
    input = wrapper.find('input');
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function))
    expect(input.prop('value')).toBe('');
  });
  
  
  
  
});
