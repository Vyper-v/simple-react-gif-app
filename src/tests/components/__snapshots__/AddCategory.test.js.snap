// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory Component tests should be rendered correctly 1`] = `
<form
  className="flex flex-col"
  onSubmit={[Function]}
>
  <input
    className="p-1 outline-1"
    onChange={[Function]}
    placeholder="Type something..."
    type="text"
    value=""
  />
</form>
`;
